packages/multichain/src/handlers/wallet-invokeMethod.ts

```diff
@@ -62,10 +62,9 @@ async function walletInvokeMethodHandler(
     return end(providerErrors.unauthorized());
   }
 
-  const scopeObject = mergeScopes(
-    caveat.value.requiredScopes,
-    caveat.value.optionalScopes,
-  )[scope];
+  const scopeObject = getSessionScopes(caveat.value)[
+    scope as InternalScopeString
+  ];
```

Comment:
Yeah, we still need to keep the interface consistent. I.e. trying to call a scope/method that wasn't returned to you via `wallet_createSession` / `wallet_sessionChanged` should throw an unauthorized error rather than a method not found error